### ISSUE TITLE
segv when stepping through code with lldb-11

### ISSUE THREAD

**Issue by mgood7123** [llvm/llvm-project#63545]
```sh
git clone --recursive https://github.com/mgood7123/MiniDoc -b 78799af1f9ccfab7352981a02a72d824e63ee950

cd MinIDoc

make build_debug && lldb ./debug_EXECUTABLE/minidoc_exe
```
```console
cd debug_BUILD ; mkdir EXECUTABLES; cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS_DEBUG="-g3 -O0" .. ; make && if test -e EXECUTABLES ; then cd EXECUTABLES; for file in * ; do mv -v $file ../../debug_EXECUTABLE/$FILE ; done ; cd ..; rmdir EXECUTABLES; fi
CMake Warning (dev) at CMakeLists.txt:24 (set):
  Cannot set "MINIDOC_INCLUDE": current scope has no parent.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
-- Generating done
-- Build files have been written to: /home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD
make[1]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[2]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[  8%] Built target StringAdapter
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[ 25%] Built target darcs_patch
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[ 33%] Built target minidoc
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[ 41%] Built target testBuilder_executables_minidoc_exe
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[ 50%] Built target testBuilder_executables_darcs_patch_exe
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[ 58%] Built target gtest
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[ 66%] Built target gmock
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[ 75%] Built target gmock_main
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[ 83%] Built target gtest_main
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[ 91%] Built target testBuilder_executables_StringAdapter_Tests
make[3]: Entering directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[3]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
[100%] Built target testBuilder_executables_DarcsPatch_Tests
make[2]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
make[1]: Leaving directory '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_BUILD'
mv: cannot stat '*': No such file or directory
(lldb) target create "./debug_EXECUTABLE/minidoc_exe"
Current executable set to '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_EXECUTABLE/minidoc_exe' (x86_64).
(lldb) b 12
Breakpoint 1: where = minidoc_exe`main + 105 at executable.cpp:12:11, address = 0x000000000000558d
(lldb) r
Process 5124 launched: '/home/DATA/git/Gecko/HTML/qparse/MiniDoc/debug_EXECUTABLE/minidoc_exe' (x86_64)
hi
sizeof MiniDoc: 880
sizeof MiniDoc::Info: 696
Process 5124 stopped
* thread #1, name = 'minidoc_exe', stop reason = breakpoint 1.1
    frame #0: 0x000055555555958d minidoc_exe`main at executable.cpp:12:11
   9   	  printf("sizeof MiniDoc::Info: %zu\n", sizeof(MiniDoc::MiniDoc_T::Info));
   10  	 
   11  	  m.load("1234");
-> 12  	  m.print();
   13  	  m.replace(1, "47");
   14  	  m.print();
   15  	  m.replace(1, "86");
(lldb) s
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
/lib/x86_64-linux-gnu/libLLVM-11.so.1(_ZN4llvm3sys15PrintStackTraceERNS_11raw_ostreamE+0x1f)[0x7f53f5bd0f8f]
/lib/x86_64-linux-gnu/libLLVM-11.so.1(_ZN4llvm3sys17RunSignalHandlersEv+0x50)[0x7f53f5bcf2f0]
/lib/x86_64-linux-gnu/libLLVM-11.so.1(+0xbd1465)[0x7f53f5bd1465]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x13140)[0x7f53fdff8140]
/lib/x86_64-linux-gnu/libclang-cpp.so.11(+0xd95199)[0x7f53faf95199]
/lib/x86_64-linux-gnu/libclang-cpp.so.11(+0xd94965)[0x7f53faf94965]
/lib/x86_64-linux-gnu/libclang-cpp.so.11(_ZNK5clang10ASTContext18getASTRecordLayoutEPKNS_10RecordDeclE+0x173c)[0x7f53faf8c8ac]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x9ef2cf)[0x7f53fd9ef2cf]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x643c0b)[0x7f53fd643c0b]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x59ecad)[0x7f53fd59ecad]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x59e303)[0x7f53fd59e303]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x5bcb60)[0x7f53fd5bcb60]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x568bc1)[0x7f53fd568bc1]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x566892)[0x7f53fd566892]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x566892)[0x7f53fd566892]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x56680f)[0x7f53fd56680f]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x6bf948)[0x7f53fd6bf948]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x6bfd65)[0x7f53fd6bfd65]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x6c49a2)[0x7f53fd6c49a2]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x6f65e6)[0x7f53fd6f65e6]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x698e9b)[0x7f53fd698e9b]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x698413)[0x7f53fd698413]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x556cc8)[0x7f53fd556cc8]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x557455)[0x7f53fd557455]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x557676)[0x7f53fd557676]
/lib/x86_64-linux-gnu/liblldb-11.so.1(+0x5fd489)[0x7f53fd5fd489]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x7ea7)[0x7f53fdfecea7]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f)[0x7f53f4f27a2f]
Segmentation fault
```

**Comment by Michael137** [comment]
Can you repro this with a more recent lldb version?

Crash doesnt repro for me with lldb-14 (and top-of-tree)

**Comment by mgood7123** [comment]
> Can you repro this with a more recent lldb version?
> 
> Crash doesnt repro for me with lldb-14 (and top-of-tree)

im using
```
lldb-11
gcc 10
llvm-11
clang 11
```

**Comment by mgood7123** [comment]
with `lldb-13` this does not crash, but with `lldb-11` this does crash

**Comment by EugeneZelenko** [comment]
@mgood7123: 11 is too old. Only most recent release (16) is maintained.

**Comment by mgood7123** [comment]
> @mgood7123: 11 is too old. Only most recent release (16) is maintained.

alright, my distro has clang/llvm/lldb 9, 11, 12 (broken package) and 13
